oak-search-elastic/src/test/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexQueryCommonTest.java

```diff
@@ -57,4 +57,26 @@ public void descendantTestWithIndexTagExplain() throws Exception {
                 + "  where isdescendantnode([nt:base], [/test]) */]", result.toString());
     }
 
+    @Override
+    public String getContainsValueFortestEqualityQuery_native() {
+        return "\"filter\":[{\"term\":{\":ancestors\":{\"value\":\"/test\",\"boost\":1.0}}},{\"term\":{\"propa.keyword\":{\"value\":\"bar\",\"boost\":1.0}}}]";
+    }
+
+    @Override
+    public String getContainsValueFortestInequalityQuery_native() {
+        return "\"filter\":[{\"term\":{\":ancestors\":{\"value\":\"/test\",\"boost\":1.0}}},{\"exists\":{\"field\":\"propa\",\"boost\":1.0}}," +
+                "{\"bool\":{\"must_not\":[{\"term\":{\"propa.keyword\":{\"value\":\"bar\",\"boost\":1.0}}}]";
+    }
+
+    @Override
+    public String getContainsValueFortestInequalityQueryWithoutAncestorFilter_native() {
+        return "\"filter\":[{\"exists\":{\"field\":\"propa\",\"boost\":1.0}},{\"bool\":" +
+                "{\"must_not\":[{\"term\":{\"propa.keyword\":{\"value\":\"bar\",\"boost\":1.0}}}]";
+    }
+
+    @Override
+    public String getContainsValueFortestEqualityInequalityCombined_native() {
+        return "\"filter\":[{\"term\":{\":ancestors\":{\"value\":\"/test\",\"boost\":1.0}}},{\"term\":{\"propb.keyword\":{\"value\":\"world\",\"boost\":1.0}}}," +
+                "{\"exists\":{\"field\":\"propa\",\"boost\":1.0}},{\"bool\":{\"must_not\":[{\"term\":{\"propa.keyword\":{\"value\":\"bar\",\"boost\":1.0}}}]";
+    }
```

Comment:
can we add one more test for the case where value is not null, or does it exist already?

Reply:
@averma21  - yes there are multiple tests. 
testEqualityQuery_native - > tests simple equality condition
testEqualityInequalityCombined_native -> equality and inequality
testInequalityQuery_native -> tests inequality with nodes created with this property and also without this property.

So effectively the condition that x<>1 should return nodes where x is set and not equal to 1 is being tested. I hope this is what you were asking for - right ?

Reply:
I was thinking of query on the lines of -
```
"select * from [nt:unstructured] where [x] is not null"
```
The execution flow of this query would also go through the changes done in PropertyValueImpl.java, right?

Reply:
Added here https://github.com/apache/jackrabbit-oak/pull/465/commits/988b46ea53eba0e523c34a319b1c914cf7422d57